app/src/main/java/com/keylesspalace/tusky/ViewMediaActivity.kt

```diff
@@ -113,13 +114,16 @@ class ViewMediaActivity : BaseActivity(), ViewImageFragment.PhotoActionsListener
             SingleImagePagerAdapter(this, imageUrl!!)
         }
 
+        hasMultiple = attachments != null && attachments!!.size > 1
+
         binding.viewPager.adapter = adapter
         binding.viewPager.setCurrentItem(initialPosition, false)
         binding.viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
             override fun onPageSelected(position: Int) {
                 binding.toolbar.title = getPageTitle(position)
             }
         })
+        binding.viewPager.isUserInputEnabled = hasMultiple
```

Comment:
At least for a single image this is a 100% improvement: have no (useless) view paging there